### PR TITLE
Adding Travis CI config

### PR DESCRIPTION
**Pull request by mattsb42-aws**
Adding Travis CI config, adding badges to readme, and updating integration test configuration methodology from config file to environment variables (primarily to simplify in running integration tests in Travis).

### CODE REVIEW

**Comment by mattsb42-aws** [comment]
Travis blocks secure environment variables in builds for pull requests from forked repositories. We'll need to decide how we want to handle that so that integration tests can pass but we're not opening ourselves to security issues.

Latest build from my fork:
https://travis-ci.org/mattsb42-aws/aws-encryption-sdk-python/builds/278062847

**Comment by mattsb42-aws** [comment]
Per offline discussion, I disabled integration tests in Travis until we have a story on how we want to handle credentials for them.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      env: TOXENV=py35
+    - python: 3.6
+      env: TOXENV=py36
+    - python: 3.6
+      env: TOXENV=bandit
+    - python: 3.6
+      env: TOXENV=doc8
+    - python: 3.6
+      env: TOXENV=readme
+# pending reorg of deserialize_header
```
Comment (lizroth) [inline .travis.yml]: As discussed, flag this as TODO and link to some tracker so we don't forget?

File: test/integration/README.rst [hunk 2]
```diff
@@ -0,0 +1,13 @@
+************************************
+aws-encryption-sdk Integration Tests
+************************************
+
+In order to run these integration tests successfully, these things which must be configured.
```
Comment (lizroth) [inline test/integration/README.rst]: Minor: wording polish ("these things which must be configured")

File: test/integration/integration_test_utils.py [hunk 3]
```diff
@@ -14,60 +14,31 @@
...
 def skip_tests():
-    blocker_var_name = 'AWS_ENCRYPTION_SDK_PYTHON_INTEGRATION_TEST_CONTROL'
-    blocker_val = os.environ.get(blocker_var_name, None)
-    if blocker_val != 'RUN':
-        return True
-    return False
+    """Only run tests if both required environment variables are found."""
+    test_control = os.environ.get(TEST_CONTROL, None)
+    key_id = os.environ.get(AWS_KMS_KEY_ID, None)
+    return not (test_control == 'RUN' and key_id is not None)
```
Comment (lizroth) [inline test/integration/integration_test_utils.py]: Minor: not a huge deal given the context, but could be a bit more readable for maintenance's sake; even just expanding the negate out would help: `test_control != 'RUN' or key_id is None` (if I did that right)

File: .travis.yml (hunk 1 above)
Comment (mattsb42-aws) [reply .travis.yml]: Issue opened to fix the underlying issue: https://github.com/awslabs/aws-encryption-sdk-python/issues/17

File: test/integration/integration_test_utils.py (hunk 3 above)
Comment (mattsb42-aws) [reply test/integration/integration_test_utils.py]: I'll be refactoring a decent chunk of the integration test utils when we get the test vectors packaged as a Python package, so I'll leave refining this bit until then.

File: test/integration/README.rst (hunk 2 above)
Comment (mattsb42-aws) [reply test/integration/README.rst]: Updated to "these things must be configured".

**Pull request merged**